graph/oval_graph.py

```diff
@@ -164,6 +164,63 @@ def save_tree_to_dict(self):
             'child': [child.save_tree_to_dict() for child in self.children]
         }
 
+    def _get_node_icon(self):
+        value = self.evaluate_tree()
+        icon = None
+        if value is None:
+            if self._is_negated_boolean('true', self.value):
+                icon = 'false'
+            elif self._is_negated_boolean('false', self.value):
+                icon = 'true'
+            else:
+                icon = self.value
+            icon, value = self.value, icon
+        else:
+            if self._is_negated_boolean('true', value):
+                icon = 'false'
+            elif self._is_negated_boolean('false', value):
+                icon = 'true'
+            else:
+                icon = value
+
+        VALUE_TO_COLOR = {
+            "true": "text-success",
+            "false": "text-danger",
+            "error": "text-dark",
+            "unknown": "text-dark",
+            "noteval": "text-dark",
+            "notappl": "text-dark"
+        }
+
+        VALUE_TO_ICON = {
+            "true": "glyphicon glyphicon-ok ",
+            "false": "glyphicon glyphicon-remove ",
+            "error": "glyphicon glyphicon-question-sign ",
+            "unknown": "glyphicon glyphicon-question-sign ",
+            "noteval": "glyphicon glyphicon-question-sign ",
+            "notappl": "glyphicon glyphicon-question-sign "
+        }
+
+        return dict(
+            color=VALUE_TO_COLOR[value],
+            icon=VALUE_TO_ICON[icon])
+
+    def to_JsTree_dict(self):
+        icons = self._get_node_icon()
+        if not self.children:
+            return {
+                'text': '<span class="' + icons['color'] + '">' + self._get_label() + '</span>',
+                "icon": icons['icon'] + icons['color'],
+                "state": {
+                    "opened": True}}
+        return {
+            'text': '<span class="' + icons['color'] + '">' + self._get_label() + '</span>',
+            "icon": icons['icon'] + icons['color'],
+            "state": {
+                "opened": True},
+            'children': [
+                child.to_JsTree_dict() for child in self.children]}
+
```

Comment:
The 2 dictionaries differ only in the `children`  key. I suggest creating only  one dictionary, then writing a condition that optionally adds the `children` key, and then a single `return` command. That way you'll prevent code duplication again.